### PR TITLE
Encapsulate evaluationMode selection (towards v5.0)

### LINKED ISSUE #747: 5.0 activation
Code for 5.0 activation for old nodes already in the interpreter:

```
if (context.activatedScriptVersion > Interpreter.MaxSupportedScriptVersion) {
        // > 90% has already switched to higher version, accept without verification
        // NOTE: this path should never be taken for validation of candidate blocks
        // in which case Ergo node should always pass Interpreter.MaxSupportedScriptVersion
        // as the value of ErgoLikeContext.activatedScriptVersion.
        // see also ErgoLikeContext ScalaDoc.
        return Success(true -> context.initCost)
      }
```
so old nodes simply skipping scripts check (downgrading to SPV security effectively)
For new (5.0) nodes, switch to JIT costing should be implemented when MaxSupportedScriptVersion increased.

For upcoming soft-forks, only adding new opcodes and types planned, thus existing soft-forking protocol would be enough?

### PR DESCRIPTION
**Pull request by aslesarenko**
Closes #747 

- [x] Evaluation mode encapsulated
- [x] Soft-fork activation tests updated
- [x] Tests cases for v6.0 activation added

### CODE REVIEW

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala [hunk 1]
```diff
@@ -366,6 +356,23 @@ trait Interpreter extends ScorexLogging {
...
+    * By default evaluation mode is determined based on `context.activatedScriptVersion`
+    * so that the interpreter works either as v4.x of v5.0.
+    *
+    * Alternatively, the required evaluation mode can be specified by giving Some(mode)
+    * value in `evalSettings.evaluationMode`, in which case the interpreter works as
+    * spedified.
+    */
+  def getEvaluationMode(context: CTX): EvaluationMode = {
+    evalSettings.evaluationMode.getOrElse {
+      if (context.activatedScriptVersion < Interpreter.MaxSupportedScriptVersion)
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:369]: It is better  to introduce a new constant like JitActivationVersion, then there is no need to check MaxSupportedScriptVersion usages before increasing it in future (which can be easily missed)

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala [hunk 2]
```diff
@@ -366,6 +356,23 @@ trait Interpreter extends ScorexLogging {
     fullJitCost
   }
 
+  /** Returns evaluation mode used by this interpreter in the given context.
+    * By default evaluation mode is determined based on `context.activatedScriptVersion`
+    * so that the interpreter works either as v4.x of v5.0.
+    *
+    * Alternatively, the required evaluation mode can be specified by giving Some(mode)
+    * value in `evalSettings.evaluationMode`, in which case the interpreter works as
+    * spedified.
```
Comment (kushti) [inline sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala:365]: spedified -> specified

Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala]: fixed

File: sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala (hunk 1 above)
Comment (aslesarenko) [reply sigmastate/src/main/scala/sigmastate/interpreter/Interpreter.scala]: done

**Review by kushti — APPROVED** [review]

**Pull request merged**